include/envoy/event/dispatcher.h

```diff
@@ -205,11 +205,14 @@ class Dispatcher : public DispatcherBase {
    * empty, it will not use any specific resolvers, but use defaults (/etc/resolv.conf)
    * @param use_tcp_for_dns_lookups if set to true, tcp will be used to perform dns lookups.
    * Otherwise, udp is used.
+   * @param no_default_search_domain_for_dns_lookups if set to true, default search domains will not
+   * be used for dns lookups.
    * @return Network::DnsResolverSharedPtr that is owned by the caller.
    */
-  virtual Network::DnsResolverSharedPtr
-  createDnsResolver(const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
-                    bool use_tcp_for_dns_lookups) PURE;
+  virtual Network::DnsResolverSharedPtr createDnsResolver(
```

Comment:
Please run `tools/code_format/check_format.py fix` before any commits.  This keeps the format consistent.

Secondly, I suggest replacing `use_tcp_for_dns_lookups` as an input parameter and source it's value from the flags.   Default to false.

Reply:
All these suggestions are noted. Now everything should be available in the updated branch.